### DISCUSSION TITLE
Cadence and predictability of sos tagging

### DISCUSSION THREAD

**Discussion by pmoravec** [sosreport/sos#3680] [Announcements]
Over the past 15-ish months, we (Red Hat folks) have been releasing upstream z-stream tags (4.7.1 or similar) at a random unpredictable manner, purely driven by our downstream needs and capabilities. That unpredictability does not look good and I apologize for any inconvenience caused by that.

After some internal discussion, we commit to the further plan: **we will create two z-stream tags per sos release, with 2 months cadence**. That means:

- sos 4.7.2 to be released  by us in a few days
- sos 4.8.0 to be released (by Jake) as usual (mid Aug)
- sos 4.8.1 to be released by us in mid Oct
- sos 4.8.2 to be released by us in mid Dec
- sos 4.9.0 to be released (by Jake) as usual (mid Feb)
- ..

Let me know if you have some question or concern to that.

**Comment by arif-ali** [comment]
Hey, thanks for your update on this.

We have been waiting for the latest z-release to come out, so that we can package that up. 

For us, it would be awesome if we can get https://github.com/sosreport/sos/pull/3646 approved and merged in time for that release.

On top of that, if this is going to be a regular thing, are we ok to add these milestones into GH, so that we know the expectations. I'd be happy to contribute and help with any of the releases. After going through our internal release process we have found changes in newer distributions that releasing the .0 release became problematic, and a newer z-release would have made it more simpler as we would not need to import extra patches in. So, I would be very much on board, and if an extra release is needed, then I would be happy to do it.

**Comment by dnegreira** [comment]
Hi @pmoravec,

Thanks for the update and willingness to keep steering the versioning in a reliable manner, this is an excellent step forward.
I would also like to propose a way for us to volunteer to do the interim releases ourselves so that we can get acquainted with the release process to grow that confidence on releasing.
This would also relieve you and Jake a bit, even though, of course, I predict lots of questions and documentation that have to be written initially.

**Comment by TurboTurtle** [comment]
> On top of that, if this is going to be a regular thing, are we ok to add these milestones into GH, so that we know the expectations.

This is my ask as well. I don't necessarily care _when_ the releases happen, just so long as we have a rough target known ahead of time so that we can make plans around them. Of course things can and will change, and the milestone dates don't need to be set in stone, but a general "we're planning a z-stream release around XYZ date" would be very helpful.

**Comment by pmoravec** [comment]
Adding milestones to GH sounds good to me. Maybe worth having a separate issue/discussion/.. for each such release to align expectations and work. Like "hey, I would like to get PR 1234 in, can you review it?" or "So I will do the tagging in a few days" (and any volunteer to this is welcomed, for sure!).

(for the upcoming 4.7.2 release/tag, until something screws up or makes me too busy, I will bundle that tomorrow)

**Comment by arif-ali** [comment]
Hey @pmoravec if you're planning a 4.8.2 in December, please tag, just so that we know the schedule, and be prepped. I'm keen to land a few items before that, so having a good date to work towards would be great. 4.8.2 is likely to be our next big release that gets propagated to all our stable releases

**Reply by pmoravec** [reply]
Hi,
thanks for reminding me. I plan doing the release/tag in mid Dec, so I created Milestone https://github.com/sosreport/sos/milestone/29 .

**Comment by pmoravec** [comment]
Upon a request from @arif-ali who would like to run some testing before the 4.8.2 release, we propose a code freeze since 9th Dec till 16th Dec / the 4.8.2 release. Just fixes preventing some regression or evident bugs shall be merged to upstream that time.

**Reply by pmoravec** [reply]
Heads up: 4.8.2 is just behind the weekend, so if you have some last simple PRs, let raise them.

**Comment by arif-ali** [comment]
I've added milestones for 4.9.1 and 4.9.2, to similar timings as last year. Happy to get these released if needed at the time. If they don't fit with RH timelines, then change them. I'm just getting ahead of the game to know when these arrive.

Hope that's ok
